test/unit/test-ini-load.js

```diff
@@ -28,57 +32,190 @@ describes.realWin('friendly-iframe-embed', {amp: true}, (env) => {
     ampdoc = env.ampdoc;
   });
 
-  it('should find and await all visible content elements in given rect', async () => {
+  function resource(win, tagName, id, overlaps = true, displayed = true) {
+    const deferred = new Deferred();
+    return {
+      element: {
+        tagName: tagName.toUpperCase(),
+        id,
+        whenLoaded: () => deferred.promise,
+      },
+      load: () => {
+        deferred.resolve();
+      },
+      isDisplayed: () => displayed,
+      overlaps: () => overlaps,
+      isFixed: () => false,
+      prerenderAllowed: () => true,
+      hasBeenMeasured: () => true,
+      hasOwner: () => false,
+      hostWin: win,
+    };
+  }
+
+  describe('whenContentIniLoadMeasure', () => {
+    it('should find and await all visible content elements in given rect', async () => {
+      let content1;
+      let content2;
+
+      const context = doc.createElement('div');
+      doc.body.appendChild(context);
+      const resources = Services.resourcesForDoc(ampdoc);
+      env.sandbox.stub(resources, 'get').returns([
+        (content1 = resource(win, 'amp-img', '1')),
+        (content2 = resource(win, 'amp-video', '2')),
+        resource(win, 'amp-img', '3', false), // resource outside rect
+        resource(win, 'amp-img', '4', true, false), // hidden resource
+        resource(win, 'amp-ad', '5'), // denylisted resource
+      ]);
+
+      let contentIniLoadComplete = false;
+      whenContentIniLoadMeasure(ampdoc, win, {}).then(() => {
+        contentIniLoadComplete = true;
+      });
+
+      ampdoc.signals().signal('ready-scan');
+      await new Promise(setTimeout);
```

Comment:
I've never seen this before for waiting a macrotask! neat

Reply:
Neither have I. This was in the tests before.